### ISSUE TITLE
Unpinned nightly rust toolchain

### ISSUE THREAD

**Issue by uncomputable** [BlockstreamResearch/rust-simplicity#211]
We should pin nightly like https://github.com/rust-bitcoin/rust-bitcoin/pull/2489.

**Comment by apoelstra** [comment]
I think you can copy the exact script. I will need to copy my github token into this repo.

**Comment by uncomputable** [comment]
We should pin all instances of nightly in CI, including the doc checker.

**Comment by apoelstra** [comment]
I have added a repo secret `APOELSTRA_CREATE_PR_TOKEN` which should be usable in CI workflows to create PRs here under my name. We can now copy the code from rust-bitcoin or rust-miniscript to pin nightly and update the pin every week.

**Comment by uncomputable** [comment]
Is it necessary to pin nightly to the latest possible version? How about we update it manually every half year / whenever we need to?

**Comment by apoelstra** [comment]
Yes, because there are constantly new clippy lints and (usually) the new ones are pretty useful for us.

**Comment by uncomputable** [comment]
Thanks for clarifying. Keeping nightly up to date seems opposite from keeping a stable, old MSRV, but I can see what we are getting.

I will assume that new clippy lints will not flag many false positives because of unfinished lint implementations.

**Comment by apoelstra** [comment]
I'd swag that there's a new useful lint every month or two and a false positive one every 3 or 4 months.

>Keeping nightly up to date seems opposite from keeping a stable, old MSRV, but I can see what we are getting

Just because we support an old MSRV doesn't mean we should let the crate break on new compilers. (And even if it did, it helps to get early warning so that we can complain when the compiler breaks us, rather than waiting years to update our MSRV then discovering it.)
